samples/vSphereAutoRestartManager.py

```diff
@@ -0,0 +1,125 @@
+            obj.append(c)
+
+        return obj
+
+    def action_hosts(self, commaList, connection, defstartdelay):
+        self.log.debug("Actioning the Provided Hosts")
+        acthosts = commaList.split(",")
+        allhosts = self.get_hosts(connection)
+
+        for h in allhosts:
+            for a in acthosts:
+                if h.name == a:
+                    self.enable_autorestart(h, defstartdelay)
+                else:
+                    self.log.info("The host \"%s\" cant be found" % a)
+
+    def enable_autorestart(self, host, defstartdelay):
+        self.log.debug("Enabling Auto Restart for Host")
+        self.log.info("The Selected Host is \n  %s" % (host.name))
+        self.log.debug("Setting the Selected Host default AutoStartManager")
+        hostDefSettings = vim.host.AutoStartManager.SystemDefaults()
+        hostDefSettings.enabled = True
+        hostDefSettings.startDelay = int(defstartdelay)
+        self.log.debug("virtual machines and applying Auto Start settings")
+        order = 1
+        for vhost in host.vm:
+            spec = host.configManager.autoStartManager.config
+            spec.defaults = hostDefSettings
+            auto_power_info = vim.host.AutoStartManager.AutoPowerInfo()
+            auto_power_info.key = vhost
+            self.log.info("The VM \"%s\" is updated if On" % (vhost.name))
+            self.log.info("VM Status is %s " % (vhost.runtime.powerState))
+            if vhost.runtime.powerState == "poweredOff":
+                auto_power_info.startAction = 'None'
+                auto_power_info.waitForHeartbeat = 'no'
+                auto_power_info.startDelay = -1
+                auto_power_info.startOrder = -1
+                auto_power_info.stopAction = 'None'
+                auto_power_info.stopDelay = -1
+            elif vhost.runtime.powerState == "poweredOn":
+                # note use of constant instead of string
+                auto_power_info.startAction = 'powerOn'
+                auto_power_info.startDelay = -1
+                auto_power_info.startOrder = -1
+                auto_power_info.stopAction = 'None'
+                auto_power_info.stopDelay = -1
+                auto_power_info.waitForHeartbeat = 'no'
+                spec.powerInfo = [auto_power_info]
+                order = order + 1
+                self.log.debug("Apply Setting to Host")
+                host.configManager.autoStartManager.ReconfigureAutostart(spec)
+
+# MAIN
+    def main(self):
+        self.log.debug("Starting")
+        self.log.debug("Getting Config")
+        self.log.debug("Connecting to vSphere")
+        connection = self.get_connection(
+            self.params.ipadd, self.params.user, self.params.password)
+
+        if self.params.listallhosts is True:
+            vSpherehosts = self.get_hosts(connection)
+            self.log.info("All the Hosts Attached are :\n")
+            for hosts in vSpherehosts:
+                print hosts.name + "\n"
```

Comment:
Shouldn't this be a log? Im basing that on what is on line 96 just above. I also think it might be better as a debug log. In a large environment you could see hundreds of hosts listed here.